digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -748,6 +749,37 @@ pub async fn get_token_supply(
     })
 }
 
+pub async fn get_token_account_balance(
+    conn: &impl ConnectionTrait,
+    token_acc_address: Vec<u8>,
+) -> Result<SolanaRpcResponseAndContext<UiTokenAmount>, DbErr> {
+    let token_acc = token_accounts::Entity::find()
+        .filter(token_accounts::Column::Pubkey.eq(token_acc_address))
+        .one(conn)
+        .await?
+        .ok_or(DbErr::RecordNotFound("Token Account Not Found".to_string()))?;
+
+    let mint_acc = tokens::Entity::find()
+        .filter(tokens::Column::Mint.eq(token_acc.mint))
+        .one(conn)
+        .await?
+        .ok_or(DbErr::RecordNotFound("Mint Account Not Found".to_string()))?;
```

Comment:
This should be queryable through sea orm relationship where token is parent of token accounts. In this case will execute a join and select all fields. sea-orm puts the query together and give back the entities.